### PR TITLE
=per #17844 Clarify docs around journal dispatchers

### PR DESCRIPTION
**Pull request by patriknw**
and never run plugin actor on default dispatcher

### CODE REVIEW

**Comment by patriknw** [comment]
Refs  #17844

**Comment by patriknw** [comment]
@ktoso can you take a look at this one, would like to merge asap, since I'm working in same files for other issue

**Comment by ktoso** [comment]
Sure! on it.

File: akka-docs/rst/java/lambda-persistence.rst [hunk 1]
```diff
@@ -688,8 +688,13 @@ A journal plugin can be activated with the following minimal configuration:
 .. includecode:: ../scala/code/docs/persistence/PersistencePluginDocSpec.scala#journal-plugin-config
 
 The specified plugin ``class`` must have a no-arg constructor. The ``plugin-dispatcher`` is the dispatcher
-used for the plugin actor. If not specified, it defaults to ``akka.persistence.dispatchers.default-plugin-dispatcher``
-for ``SyncWriteJournal`` plugins and ``akka.actor.default-dispatcher`` for ``AsyncWriteJournal`` plugins.
+used for the plugin actor. If not specified, it defaults to ``akka.persistence.dispatchers.default-plugin-dispatcher``.
+
+The journal plugin instance is an actor so the methods corresponding to requests from persistent actors
+are executed sequentially. It may delegate to asyncrounous libraries, spawn futures, or delegate to other
```
Comment (ktoso) [inline akka-docs/rst/java/lambda-persistence.rst]: asynchronous

File: akka-docs/rst/java/lambda-persistence.rst [hunk 2]
```diff
@@ -705,6 +710,12 @@ A snapshot store plugin can be activated with the following minimal configuratio
 The specified plugin ``class`` must have a no-arg constructor. The ``plugin-dispatcher`` is the dispatcher
 used for the plugin actor. If not specified, it defaults to ``akka.persistence.dispatchers.default-plugin-dispatcher``.
 
+The snapshot store instance is an actor so the methods corresponding to requests from persistent actors
+are executed sequentially. It may delegate to asyncrounous libraries, spawn futures, or delegate to other
```
Comment (ktoso) [inline akka-docs/rst/java/lambda-persistence.rst]: same typo

File: akka-docs/rst/scala/persistence.rst [hunk 3]
```diff
@@ -652,7 +652,7 @@ A Simple Example
 To demonstrate the features of the ``PersistentFSM`` trait, consider an actor which represents a Web store customer.
 The contract of our "WebStoreCustomerFSMActor" is that it accepts the following commands:
 
-.. includecode:: ../../../akka-persistence/src/test/scala/akka/persistence/fsm/PersistentFSMSpec.scala#customer-commands
+.. includecode:: ../../../akka-persistence/src/test/scala/akka/persistence/fsm/PersistentFSMActorSpec.scala#customer-commands
```
Comment (ktoso) [inline akka-docs/rst/scala/persistence.rst]: Renaming the file would be preferable. The class is now called `PersistentFSM`

Comment (ktoso) [inline akka-docs/rst/scala/persistence.rst]: (good catch, sorry If I introduced this mistake)

File: akka-persistence/src/main/resources/reference.conf [hunk 4]
```diff
@@ -71,6 +72,7 @@ akka.persistence {
                 parallelism-max = 8
             }
         }
+        # Default dispatcher for streaming snapshot IO
         default-stream-dispatcher {
```
Comment (ktoso) [inline akka-persistence/src/main/resources/reference.conf]: is this used?

Comment (ktoso) [inline akka-persistence/src/main/resources/reference.conf]: hm ok, it is in config `stream-dispatcher = "akka.persistence.dispatchers.default-stream-dispatcher"`; nevermind, keep as is

**Comment by ktoso** [comment]
LGTM after typo fixed (or merge as is and fix typo in next PR)

File: akka-persistence/src/main/resources/reference.conf (hunk 4 above)
Comment (patriknw) [inline akka-persistence/src/main/resources/reference.conf]: it's a bit weird, I'll move it into the file snapshot config, as that is the only place it is used

Comment (patriknw) [inline akka-persistence/src/main/resources/reference.conf]: changed my mind, leaving this as is

File: akka-docs/rst/java/lambda-persistence.rst (hunk 1 above)
Comment (patriknw) [inline akka-docs/rst/java/lambda-persistence.rst]: thx

File: akka-docs/rst/scala/persistence.rst (hunk 3 above)
Comment (patriknw) [inline akka-docs/rst/scala/persistence.rst]: now it is PersistentFSMSpec.scala

File: akka-persistence/src/main/resources/reference.conf (hunk 4 above)
Comment (ktoso) [inline akka-persistence/src/main/resources/reference.conf]: okey

**Pull request merged**